internal/p2p/router.go

```diff
@@ -642,6 +639,7 @@ func (r *Router) connectPeer(ctx context.Context, address NodeAddress) {
 
 	if err := r.runWithPeerMutex(func() error { return r.peerManager.Dialed(address) }); err != nil {
 		r.logger.Error("failed to dial peer", "op", "outgoing/dialing", "peer", address.NodeID, "err", err)
+		r.peerManager.dialWaker.Wake()
```

Comment:
I think it's a problem if we fail to connect to a peer and *don't* call the wakeup thread, this is a case where we would otherwise shed peers and not push a new peer for dialing